Engine/TestPlanExecution.cs

```diff
@@ -180,6 +180,14 @@ failState execTestPlan(TestPlanRun execStage, IList<ITestStep> steps)
             Stopwatch planRunOnlyTimer = Stopwatch.StartNew();
             var runs = new List<TestStepRun>();
 
+            bool didBreak = false;
+            void addBreakResult()
+            {
+                if (didBreak) return;
+                didBreak = true;
+                execStage.Parameters.Add(new ResultParameter("CancelledDueToBreakCondition", "Yes")); 
```

Comment:
Lets move this property name to a static and internal variable in TestPlanRun. Also we could use the GUID of the step causing the plan to cancel?

Reply:
Good call!